### ISSUE TITLE
Ranges can only match single chars (SC2102)

### ISSUE THREAD

**Issue by masavini** [koalaman/shellcheck#2113]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2102
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#! /bin/bash

[[ -v ARR[index] ]]
[[ -v 'ARR[index_with_underscore]' ]]
[[ -v ARR[index_with_underscore] ]]

var='bubu'
printf -v ARR[index] '%s' "${var}"
printf -v 'ARR[index_with_underscore]' '%s' "${var}"
printf -v ARR[index_with_underscore] '%s' "${var}"

```

#### Here's what shellcheck currently says:
Line 5:
[[ -v ARR[index_with_underscore] ]]
         ^-- SC2102: Ranges can only match single chars (mentioned due to duplicates).
 
Line 10:
printf -v ARR[index_with_underscore] '%s' "${var}"
             ^-- SC2102: Ranges can only match single chars (mentioned due to duplicates).

#### Here's what I wanted or expected to see:
Nothing.

**Comment by koalaman** [comment]
Thanks! This issue is now fixed, see #2285.
